### PR TITLE
6472- backend changes

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/6472

*Description of changes:*

- Added migration script for adding suspensionreasoncodes table creation and seeding
- Added foreign key realtionship with orgs
- Added mdoel and schema model for suspensionreasoncode 
- Added logic for patch operation when status change to suspend/active in org service

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Review by mengdong19 — CHANGES_REQUESTED** [review]
Ideally, we'll need to include suspension reason code in org responses, so it could be read for front end use, org response schemas need to be updated accordingly.

File: auth-api/migrations/versions/5053985bdfc6_added_suspension_reason_codes.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    op.add_column('orgs', sa.Column('suspension_reason_code', sa.String(length=15), nullable=True))
+    op.create_foreign_key('orgs_suspension_reason_code_fkey', 'orgs', 'suspension_reason_codes', ['suspension_reason_code'], ['code'], ondelete='SET NULL')
+    op.add_column('orgs_version', sa.Column('suspension_reason_code', sa.String(length=15), autoincrement=False, nullable=True))
+    # ### end Alembic commands ###
+
+    # Insert code values
+    op.bulk_insert(
+        suspension_reason_codes,
+        [
+            {'code': 'CHANGE', 'description': 'Account Ownership Change', 'default': True},
```
Comment (sumesh-aot) [inline auth-api/migrations/versions/5053985bdfc6_added_suspension_reason_codes.py:42]: May be change CHANGE to OWNER_CHANGE

File: auth-api/tests/unit/services/test_org.py [hunk 2]
```diff
@@ -42,7 +42,9 @@
 from auth_api.services.entity import Entity as EntityService
```
Comment (sumesh-aot) [inline auth-api/tests/unit/services/test_org.py:42]: Add a test for API

**Review by sumesh-aot — APPROVED** [review]

**Comment by karthik-aot** [comment]
6472 - backend changes : Added suspension_reason_code and relationships

**Pull request merged**